### DISCUSSION TITLE
How to run flatpak mpv as external player

### DISCUSSION THREAD

**Discussion by bheeshmpita** [FreeTubeApp/FreeTube#5501] [Q&A] [answered]
I want to use flatpak mpv as external player for streaming videos so what should be the configuration in the settings. I have gone through an already discussed similar discussion but that was in context of both the flatpak apps: freetube as well as mpv, while in my case mpv is the only flatpak app.
Freetube v0.21.2 Beta
mpv v0.38.0

**Comment by failafail8** [comment]
did you find a solution ? i tried distro+ flatpak mpv. i tried other players aswell, i tried few settings trough flatseal but i didnt managed to do it.

**Reply by bheeshmpita** [reply]
~~No, no solutions yet.~~

**Reply by xyhbhtt** [reply]
Have a look a this thread: https://github.com/FreeTubeApp/FreeTube/issues/6850#issuecomment-2675434520

**Comment by jn64** [comment, marked as answer]
This has worked all along. It can only be a configuration error.

To be clear, you want to use **non-flatpak FreeTube** + **flatpak mpv**

> [!IMPORTANT]
> In non-flatpak FreeTube:
> 
> 1. Exernal Player: mpv
> 2. Custom External Player Executable: `flatpak`
> 3. Custom External Player Args:
>    - `run`
>    - `io.mpv.Mpv`

---

That's all. The custom executable and args are what you would use to [run any flatpak](https://docs.flatpak.org/en/latest/using-flatpak.html#running-applications), unrelated to FreeTube:

```
flatpak run io.mpv.Mpv
```

You should _not_ change any permissions for io.mpv.Mpv via Flatseal (or flatpak cli) for the purpose of making it work with FreeTube. If you made random changes, you can reset them in Flatseal.

---

For reference, other use cases:

- [flatpak FreeTube + flatpak player](https://github.com/FreeTubeApp/FreeTube/issues/6850#issuecomment-2675232688)
- [flatpak FreeTube + non-flatpak player](https://github.com/FreeTubeApp/FreeTube/issues/6850#issuecomment-2674833893)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
inquires regarding external players migrated to https://github.com/FreeTubeApp/FreeTube/discussions/7789
